core/arch/arch.h

```diff
@@ -723,7 +723,7 @@ mangle_writes_thread_register(dcontext_t *dcontext, instrlist_t *ilist, instr_t
 /* offsets within local_state_t used for specific scratch purposes */
 enum {
     /* ok for this guy to overlap w/ others since he is pre-cache */
-    FCACHE_ENTER_TARGET_SLOT = TLS_REG0_SLOT,
+    FCACHE_ENTER_TARGET_SLOT = TLS_REG2_SLOT,
```

Comment:
I think the comment says that other slots can use the one that `FCACHE_ENTER_TARGET_SLOT` does.

So instead of replacing, I just added that `FCACHE_ENTER_TARGET_SLOT` cannot use reg0 and reg1 slots.